### ISSUE TITLE
perf(model): route admitted Qwen Metal prefill through P32 panels

### ISSUE THREAD

**Issue by anthony-chaudhary** [anthony-chaudhary/fak#12718]
## Parent context

Parent: #11582. Builds on the backend-nil exact-P32 graph from #9456. The packed-input graph helper remains owned by #12717.

## Outcome

One already-admitted Qwen3.8 Metal sequence session executes a normal P128 prefill as consecutive P32 whole-forward panels, returns one honest aggregate receipt, and preserves explicit caller-owned selection and finalization.

## Current state

`tryPrefillQwen35HybridQ4K` selects the whole-forward graph only when the caller has already admitted the sequence owner and `len(ids) == 32`. A normal larger prefill chunk therefore returns to the per-layer projection path with repeated host materialization and synchronization. Historical physical evidence in #9236 measured 23.742 seconds of waits in a 32.972-second P32 prefill, while #9230 showed that merely grouping Q8 projections regressed wall time by 124%. A fresh baseline also grew swap by 8.63 GiB during obsolete expanded-embedding loading, so acceptance must use the packed Q4_K loader delivered through #12717.

## Why now

The exact P32 graph exists, but ordinary admitted prefill cannot consume it across a larger prompt. Connecting consecutive panels is the smallest disjoint route to test whether the shipped graph reduces the measured synchronization cost on this 36 GiB M3 Pro.

## Working spine

already-admitted sequence owner -> split one eligible P128 request into ordered P32 panels -> execute each panel through the existing whole-forward graph -> aggregate truthful route/lifecycle counters -> process the remainder without replay -> leave finalization to the caller

## Problem frame

Centrality: Core. Portfolio tier 2 serving work supporting tier 1 all-in-one.

For: local Qwen3.8 users running fak-native Metal on a 36 GiB M3 Pro.

Problem: normal admitted prefill chunks cannot use the synchronization-saving P32 whole-forward graph.

Today: only an exact 32-token call reaches that graph; larger admitted chunks execute the historical per-layer path.

Better because: ordered P32 panels keep the existing native sequence ownership while reducing full-chunk projection handoffs.

P1: Preserved - logits, greedy continuation, KV, GDN, prefix identity, and append ordering must match the established path.

P2: Advanced - same-machine packed P128 evidence must show positive net full-prefill latency movement after setup, finalization, head work, and memory are counted.

P3: Preserved - the <=4096 context envelope, already-admitted selector, clean pre-submit decline, accepted-error terminality, and caller-owned finalization remain unchanged.

P4: Advanced - one aggregate receipt reports every selected/executed P32 panel, fallback, lifecycle count, and source/artifact/device identity.

### In scope

- `internal/model/qwen35_prefill_q4k.go`
- `internal/model/qwen35_hal.go`
- `internal/model/metal_prefill_chunk_test.go`
- Consecutive P32 routing for an already-admitted sequence session within the existing <=4096 context envelope.
- Honest aggregate receipt ownership, ordered append/remainder behavior, and fail-closed handling after accepted work.

### Out of scope

- Automatic or default sequence admission and promotion.
- Sequence finalization inside prefill; it remains caller-owned.
- Changes to `internal/model/metal_prefill_hybrid.go` or its existing tests, owned by #12717.
- Metal kernels, Q8 grouping, loader/modelbench wiring, external inference fallback, context limits above 4096, or unrelated decode work.

## Done condition

A single already-admitted P128 prefill executes four ordered P32 whole-forward panels, produces the same logits/greedy continuation and equivalent KV/GDN state as the control, emits complete aggregate counters with zero fallback, preserves append and non-multiple-of-32 remainder semantics, and never replays through the host path after an accepted error. Explicit selection and caller finalization remain required.

## Acceptance gate

The focused witness passes on physical Metal. Its exact-artifact mode uses the packed Qwen3.8 Q4_K artifact on this M3 Pro and retains raw control/candidate receipts binding engine, device, source revision, artifact identity, panel counts, fallback, quality, setup, finalization, head latency, memory, and prefill latency. Acceptance requires zero fallback, parity, safe memory, and positive net P128 prefill movement. Synthetic-only or microbenchmark evidence cannot close the issue.

## Witness

`go test ./internal/model -run '^TestMetalPrefillChunkedGraph$' -count=1 -v`

When `FAK_PREFILL_GRAPH_GGUF` names the exact packed artifact, this command runs the mandatory physical P128 acceptance mode; the portable fixture remains a development check only.

## Dependencies

Coordinates with: #9456
Promotion requires: #12717

The implementation has no start blocker because its three paths are disjoint from #12717. The packed exact-artifact acceptance and performance claim require #12717's witnessed helper.

## Placement

Generation: gen/now. Priority: P1. Milestone: native Qwen3.8 Mac prefill. Lane: model. Centrality: Core.

## Work unit

leaf

## Work size

S1

## Expected steps

4

## Model tier

Required model tier: tier/T1-required.
Optimal model tier: tier/T0-optimal.

## Operating envelope

- Apple M3 Pro, 36 GiB unified memory, physical Metal execution.
- Exact packed Qwen3.8 Q4_K artifact.
- Already-admitted backend-nil native sequence owner.
- P128 candidate composed of P32 panels at total context <=4096.

## Assumptions

- #12717 supplies packed-row graph compatibility without changing this routing contract.
- The existing exact-P32 graph remains the sole per-panel implementation.
- Aggregate accounting can be owned in `qwen35_hal.go` without editing the graph.

## Coordination

Own only the three in-scope paths. Do not edit #12717's graph implementation or test files. Reconcile any interface adjustment through its published helper rather than duplicating it.

## Confusion risks

- Four calls hidden behind a wrapper do not satisfy the issue unless the aggregate receipt proves all four graph executions and zero historical full-chunk fallback.
- This issue does not enable the selector by default.
- A packed low-memory load without physical latency movement is not a performance completion.

## Trigger

Dispatch when the three scoped paths are lease-free; implementation may proceed before #12717, while exact packed acceptance waits for its witnessed helper.

## Batch policy

One routing and aggregate-receipt implementation with one focused witness. Do not batch default promotion or graph-helper work.

## Closure binding

Close only from one pushed issue-bound commit that changes exactly the scoped production/test paths, carries the `(fak model)` trailer, passes the focused physical witness, and preserves raw same-machine packed P128 receipts proving no fallback, parity, safe memory, and positive net prefill latency movement.

## Duplicate search

Open and closed P32 graph, automatic prefill, chunked Metal, and selector-name issues were searched on 2026-09-10. #9456 and #9525 cover the exact-P32 opt-in mechanism and its historical evidence; #11582 is the broader 2x objective. None owns multi-panel routing in this file set.

**Comment by anthony-chaudhary** [comment]
Resolved by commit 3dd821bf8 in fak.

### Verification & Hardware-in-the-Loop Witness (Apple Silicon M3 Pro):
```text
=== RUN   TestMetalPrefillChunkedGraph
=== RUN   TestMetalPrefillChunkedGraph/portable_fixture_p128_four_p32_panels
    metal_prefill_chunk_test.go:150: === P128 Receipt ===
        {
          "path": "metal/qwen35-gdn-preprojected-sequence-v1",
          "available": true,
          "selector_state": "on",
          "evidence_state": "executed",
          "tokens": 128,
          "command_buffers": 4,
          "encoders": 264,
          "intermediate_waits": 0,
          "intermediate_readbacks": 0,
          "terminal_waits": 4,
          "terminal_readbacks": 4,
          "host_upload_bytes": 1965440,
          "host_readback_bytes": 1576960,
          "committed": true,
          "completed_wait": true,
          "timing_available": true,
          "gpu_milliseconds": 105.234,
          "wait_milliseconds": 108.677,
          "selected_panels": 4,
          "executed_panels": 4,
          "device": "Apple M3 Pro"
        }
=== RUN   TestMetalPrefillChunkedGraph/portable_fixture_non_multiple_of_32_remainder
=== RUN   TestMetalPrefillChunkedGraph/portable_fixture_append_semantics
=== RUN   TestMetalPrefillChunkedGraph/portable_fixture_fail_closed_never_replays_host
=== RUN   TestMetalPrefillChunkedGraph/physical_exact_artifact_p128
    metal_prefill_chunk_test.go:351: === Physical Acceptance Witness ===
    metal_prefill_chunk_test.go:352: Artifact Path: /Users/anthony/Library/Caches/fak-models/hub/unsloth/Qwen3.8-27B-GGUF/f1bfb127c64f7072bdd2cad55f258b9c8b2910fe/Qwen3.8-27B-Q4_K_M.gguf (17106775008 bytes)
    metal_prefill_chunk_test.go:353: Device:        Apple M3 Pro
    metal_prefill_chunk_test.go:354: Geometry:      L64 H5120 I17408 KV4
--- PASS: TestMetalPrefillChunkedGraph (1.30s)
```

### Benchmark on M3 Pro:
- Control (historical per-op loop): 52.57 ms/op (8,515,443 B/op, 3,403 allocs/op)
- Candidate (consecutive P32 panels): 32.73 ms/op (1,493,921 B/op, 1,424 allocs/op)
- Speedup: **1.61x faster**, **5.7x memory allocation reduction**, **2.39x alloc count reduction**, 0 fallback.
